### PR TITLE
Fix undefined variable `versionNumberRegex`

### PR DESCRIPTION
**Pull request by fer-av**
versionNumberRegex Not define!!!

| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | No anda Expert mode -> Canal: directorio local
| Type?             | bug fix 
| BC breaks?        | yes 
| Deprecations?     | yes 
| Fixed ticket?     | 
| Sponsor company   | eflyshop
| How to test?      | hice el cambio y actualice mi sitio!

### CODE REVIEW

**Comment by ps-jarvis** [comment]
Hello @fer-av!

This is your first pull request on [autoupgrade](https://github.com/PrestaShop/autoupgrade/) repository of the PrestaShop project.

Thank you, and welcome to this Open Source community!

File: js/upgrade.js [hunk 1]
```diff
@@ -777,6 +777,7 @@ $(document).ready(function () {
       var params = {};
       var $newChannel = $("select[name=channel] option:selected").val();
       var $oldChannel = $("select[name=channel] option.current").val();
+      var versionNumberRegex = /^\d+(.\d+){2,3}$/; // versionNumberRegex no definada, la agregue
```
Comment (Seb33300) [inline js/upgrade.js]:
```suggestion
      var versionNumberRegex = /^\d+(.\d+){2,3}$/;
```

**Review by Seb33300 — CHANGES_REQUESTED** [review]

**Comment by M0rgan01** [comment]
@fer-av Hello! Thanks for the contribution, and for restoring the missing variable!

 However, the upgrade.js file will be completely refactored into typescript for the introduction of a new UI on the next major release.
 
On the other hand it is possible to target the 6.x branch, which may have minor or patch releases in the future.

**Comment by Quetzacoalt91** [comment]
This variable is indeed not used anymore in the upgrade.js file on the `dev` branch (https://github.com/PrestaShop/autoupgrade/pull/902/files#diff-35ef954ff7799e931a3e99706a3bca463d56a3c77729d5755311b9ebcb93c1acL837), and the incoming version will provide a new UI.

I changed the base branch for 6.x but it requires a rebase on your end.

**Review by Quetzacoalt91 — APPROVED** [review]

**Review by AureRita — APPROVED** [review]
Hi @fer-av 

Thank you for your PR, I tested it and it seems to works as you can see : 

[recording(13).webm](https://github.com/user-attachments/assets/dfd260d9-1c3e-4e71-8eb5-9b800b261355)


Tested from : 
8.0.4 to 8.2 
8.2 to 9.0.0
8.0.4 to 9.0.0

Because the PR seems to works as expected, It's QA :heavy_check_mark: 

Thank you

**Pull request merged**